src/redux/wanted/wantedSlice.js

```diff
@@ -0,0 +1,98 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import axios from 'axios';
+import { BASE_URL } from '../constants';
+
+const initialState = {
+  wantedList: [],
+  status: 'idle',
+  current: null,
+  filtered: null,
+  error: null,
+};
+
+export const fetchWanted = createAsyncThunk('GET_WANTED', async () => Promise.all(
+  Array.from({ length: 48 }, (_, k) => k + 1).map(async (page) => {
+    const response = await axios.get(`${BASE_URL}`, { params: { page } });
+    return response.data;
+  }),
+));
+
+export const wantedSlice = createSlice({
+  name: 'wanted',
+  initialState,
+  reducers: {
+    getSingleWanted: (state, action) => {
+      const single = state.wantedList.find(
+        ({ wanted_id }) => wanted_id.toString() === action.payload.toString(),
+      );
+      state.current = single;
+    },
+    searchWanted: (state, action) => {
+      if (action.payload !== '' && action.payload !== null) {
+        const wanted = state.wantedList.filter(({ title }) => title.includes(action.payload.toUpperCase()));
+        state.filtered = wanted;
+      } else {
+        state.filtered = null;
+      }
+    },
+  },
+  extraReducers(builder) {
+    builder
+      .addCase(fetchWanted.pending, (state, _) => {
+        state.status = 'loading';
+      })
+      .addCase(fetchWanted.fulfilled, (state, action) => {
+        state.status = 'succeeded';
+        const wanted = action.payload
+          .map(({ items }) => items.map((wanted) => ({
+            aliases: wanted.aliases,
+            age_max: wanted.age_max,
+            age_min: wanted.age_min,
+            age_range: wanted.age_range,
+            complexion: wanted.complexion,
+            caution: wanted.caution,
+            coordinates: wanted.coordinates,
+            dates_of_birth_used: wanted.dates_of_birth_used,
+            details: wanted.details,
+            eyes: wanted.eyes,
+            hair: wanted.hair,
+            images: wanted.images,
+            height_max: wanted.height_max,
+            height_min: wanted.height_min,
+            nationality: wanted.nationality,
+            occupations: wanted.occupations,
+            place_of_birth: wanted.place_of_birth,
+            reward_max: wanted.reward_max,
+            reward_min: wanted.reward_min,
+            reward_text: wanted.reward_text,
+            sex: wanted.sex,
+            subjects: wanted.subjects,
+            title: wanted.title,
+            warning_message: wanted.warning_message,
+            weight: wanted.weight,
+            wanted_id: wanted.uid,
+            description: wanted.description,
+          })))
+          .flat()
+          .filter(({ reward_max }) => reward_max > 0);
+        // .sort((a, b) => b.reward_max - a.reward_max);
```

Comment:
* Kindly remove all commented code as it makes your codebase bulky and unprofessional